Flexbox lesson7

annasvrv opened the pull request.

annasvrv committed:
Styled workflow card list in homepage.

Added counter list and background images. Set position.

annasvrv committed:
Added progress effect to roadmap phase list

annasvrv committed:
Positioned details faq

annasvrv committed:
Added new div with class wrapper

annasvrv committed:
Added new div with class wrapper

annasvrv committed:
Added new div with class wrapper, refactored dl lists

annasvrv committed:
Added new div with class wrapper

annasvrv committed:
Added new div with class wrapper

annasvrv committed:
Added new div with class wrapper

annasvrv committed:
Added new div with class wrapper

annasvrv committed:
Added new div with class wrapper

annasvrv committed:
Added flexbox styles

annasvrv committed:
fixed classes

annasvrv committed:
Added background decorations - stars

annasvrv committed:
Redundant comments cleanup

alex-koshara commented:
I know that you have a static website and what happened in the screenshot does not apply to homework. As an option, I can only recommend here to finalize those points that break down. Such as:
- the text ran over the logo
- the arrow icon has moved away from its link
- the theme icon is stuck closely to the link
- and the connect button sticks to the right edge, although there is padding there
- and if you want, you could recheck all site for these cases

Lovely reminder, it's just a recommendation, it's not necessary to fix it
<img width="1073" alt="Снимок экрана 2024-08-25 в 8 08 43 AM" src="https://github.com/user-attachments/assets/3e05f5b7-8a5d-4be7-9b30-46f4c7291710">

alex-koshara commented on css/style.css:
```diff
@@ -59,7 +59,13 @@ ul {
 
 .container {
-  padding: 0 255px;
+  padding: 0 20px;
```
As a suggestion, in the future, you can combine styles for container and wrapper.

[for example](https://codepen.io/Rednaxxxela/pen/EzjwGP)

annasvrv replied:
Do you mean to add padding to wrapper and adjust max-width accordingly? 
`.wrapper {
  max-width: 1410px (+40px);
  width: 100%;
  margin: auto;
  padding-inline: 20px;
}`

alex-koshara replied:
Yep, like this!
Sorry for late answer, I was sick. Tomorrow I will answer on the rest questions and I saw you did also the grid outline!

alex-koshara commented on css/style.css:
```diff
@@ -59,7 +59,13 @@ ul {
 
 .container {
-  padding: 0 255px;
+  padding: 0 20px;
```
You can use here logical property: 
padding-inline: 20px;

annasvrv replied:
#fixed

alex-koshara commented on css/style.css:
```diff
@@ -133,23 +137,31 @@ button {
 }
 
 /************* Header navigation styles *******************/
+.nav-menu {
```
As an additional improvement, it will also be possible to think about how, when adding additional menu items, everything will also be displayed beautifully
<img width="1229" alt="image" src="https://github.com/user-attachments/assets/f8655b97-07dd-440d-856f-fc1b2c06005f">

annasvrv replied:
Not sure if it's beautiful solution... but I tried to adjust the code.

#fixed

alex-koshara commented on css/style.css:
```diff
@@ -133,23 +137,31 @@ button {
 
 /************* Header navigation styles *******************/
+.nav-menu {
+  max-width: 709px;
```
I understand that this value is taken from the layout. I would recommend rounding to multiples of 2 or 5 in the future. For example, 710px. Or make the width at the expense of the content

annasvrv replied:
#fixed

alex-koshara commented on css/style.css:
```diff
@@ -255,26 +264,53 @@ button {
 
 .hero-content {
   width: 750px;
+  padding-top: 141px;
```
Another kindly reminder that you can round the values to 2 or 5

annasvrv replied:
#fixed

alex-koshara commented on css/style.css:
```diff
@@ -291,6 +327,31 @@ button {
 }
 
+.hero-counter {
+  display: grid;
```
It's really cool that grids are used here!

What can be improved here:
1) Try to avoid use numbers in class names (hero-counter1, hero-counter2, etc)
2) add a few more definitions with values and see what happens to the markup

As we can see, this is not provided for in the styles now. In real life, there can be any number of content, and in the future our task will be to provide for this and make sure that nothing breaks when adding it.

If you have time, try to improve this moment through grids as well.

annasvrv replied:
#fixed

alex-koshara commented on css/style.css:
```diff
@@ -309,27 +370,34 @@ button {
+
+/*******************  nft-cards section ****************/
 
 .nft-cards {
```
Please, recheck first slide, I didn't see it on the 1440px resolution
<img width="1433" alt="image" src="https://github.com/user-attachments/assets/e9aea3ef-fb17-41ac-b0ec-e84f2296210b">

annasvrv replied:
#fixed

alex-koshara commented on css/style.css:
```diff
@@ -309,27 +370,34 @@ button {
 
-/**********************  nft-cards section **************************/
+.hero-image {
+  position: relative;
```
I would suggest using a wrapper for the image and moving it through the padding as needed. Very often we have to work with images and positioning. To do this, an image wrapper is also often used, which is responsible for the location, and the image is already responsible for the image itself

annasvrv replied:
#fixed

alex-koshara commented on css/style.css:
```diff
@@ -309,27 +370,34 @@ button {
   white-space: nowrap;
 }
 
 .nft-card {
```
Please note that not all of the card is a link.
<img width="319" alt="image" src="https://github.com/user-attachments/assets/771ceec2-c046-4b5f-882e-0fb30db668aa">

annasvrv replied:
#fixed

alex-koshara commented on css/style.css:
```diff
@@ -309,27 +370,34 @@ button {
-  margin: 0 auto;
   width: 330px;
   padding: 20px;
+  flex-basis: 330px;
```
Do you really need this property?

annasvrv replied:
probably not

#fixed

alex-koshara commented on css/style.css:
```diff
@@ -341,18 +409,37 @@ button {
   text-transform: uppercase;
```
Please note that there is a strange indentation between the image and the text. It would be cool if you can figure it out and make this indentation predictable.
<img width="332" alt="Снимок экрана 2024-08-25 в 11 03 27 AM" src="https://github.com/user-attachments/assets/80e86cd7-b1f5-408d-b66b-ff95f64b17b9">

Hint:
There are several solutions, it is worth paying attention to either the <a> tag or the <img>

annasvrv replied:
#fixed

alex-koshara commented on css/style.css:
```diff
@@ -363,16 +450,18 @@ button {
 }
 
 .about-list {
```
<img width="1205" alt="image" src="https://github.com/user-attachments/assets/d4c5995a-e3eb-4144-8f49-fad3953a4159">

I see that margin is hitting the button, which looks very strange. Also, the button does not fit completely into class="about" and is partially located in the padding. It would be great if you could figure out why this is happening and fix it.

<img width="544" alt="image" src="https://github.com/user-attachments/assets/1f6e5929-19ed-4aa3-b7c0-0253ae516486">

annasvrv replied:
#fixed

alex-koshara commented on css/style.css:
```diff
@@ -1441,8 +1798,84 @@ input::placeholder::first-letter {
+  color: var(--text-color);
+}
+
+.about-list-description::first-letter {
```
There is no information about this class in diff. I would like to leave some advice here for the future:
1) Try not to use a fixed width inside the block, if this can be achieved through paddings. If we delete it, nothing will happen 
<img width="1106" alt="Снимок экрана 2024-08-25 в 11 24 35 AM" src="https://github.com/user-attachments/assets/c5edc14e-03b8-4890-8fe2-c50278149b55">

2) Try not to set the height for the blocks unnecessarily. It is desirable that the height is always at the expense of the content. If we add more text now, something will break

<img width="600" alt="image" src="https://github.com/user-attachments/assets/14abce98-bc33-490c-af13-05888343df3e">

annasvrv replied:
#fixed

alex-koshara commented on css/style.css:
```diff
@@ -448,93 +537,143 @@ button {
   background-repeat: no-repeat;
-  background-position: top 10px center;
-} */
+  background-position: top 15px center;
```
It's very cool that I positioned 3 elements the same way. But something went wrong with this one. Let's try to improve it?
I found such a pattern that we always have 30px from the icon to the top of the container.
How would I do it: put this image in a pseudo-element, which would have a background in the form of these icons and these icons would be placed in the center of this pseudo-element. In this case, I always had a padding at the container at 30px, centering the image inside the pseudo-element and after that the indentation is the same everywhere at 20px before the STEP header

Please note that there may also be a mistake in the layout. Step 4 is not in line with the other steps
<img width="1191" alt="Снимок экрана 2024-08-25 в 11 37 25 AM" src="https://github.com/user-attachments/assets/8aff0f96-f064-4c14-aabf-a26dd59e9b86">

annasvrv replied:
Not sure I understood your advice correctly. I tried with pseudo-element but i cant use it in list item as it effects hover-effect. So I adjusted my code a bit as I could..

alex-koshara commented on css/style.css:
```diff
@@ -448,93 +537,143 @@ button {
 .roadmap-img-box {
   width: 715px;
+  display: grid;
+  grid-template-columns: 1fr 0.5fr;
```
A wonderful application of grids!

alex-koshara commented on css/style.css:
```diff
@@ -546,83 +685,137 @@ button {
+
+/******** Phase line **********/
 
 .roadmap-nav {
```
Advice for the future. Try to avoid this behavior when something falls out of the parent block. This behavior can lead to a breakdown of the layout if we add something else.

<img width="1261" alt="Снимок экрана 2024-08-25 в 11 44 14 AM" src="https://github.com/user-attachments/assets/72a3ef06-b7b2-46dd-be5e-cf10e10fa356">

annasvrv replied:
(I have fixed it by adding height) - ignore this please.

I have tried another solution and hope its better and do not have height restrictions.

alex-koshara commented on css/style.css:
```diff
@@ -652,44 +845,76 @@ button {
 
+.faq-box {
+  position: relative;
+  top: 250px;
```
Can you please explain why you use a negative margin in the `faq-box-section` and here the position top is 250px?

annasvrv replied:
Because I didn't find a better solution how to put  .faq-box between two backgrounds and prevent covering  next section (.cta-homepage section) when details expand by pressing +. 
I have moved negative margin to .faq-box. 
But I'll be happy to learn a better way ))

alex-koshara replied:
Maybe I'm missing something. Usually, if we have one background running over another, then we really use a negative margin. That is, we don't need the top value here. Then the cta-homepage will not need such a large padding, since everything will follow each other. And after that you need to add only necessary paddings or margins between the sections

<img width="1295" alt="Снимок экрана 2024-09-02 в 2 52 29 PM" src="https://github.com/user-attachments/assets/b028517e-6166-4c64-b355-8ff2711122b8">
<img width="1390" alt="image" src="https://github.com/user-attachments/assets/fc731f0e-baab-4287-b77c-871eaeb6c462">

alex-koshara commented on css/style.css:
```diff
@@ -652,44 +845,76 @@ button {
   width: 930px;
+}
+
+.faqs-card:not(:last-child) {
```
It's very cool that you use the pseudo-class connection here and due to this you get more experience about the ability to indent between elements!

alex-koshara commented on css/style.css:
```diff
@@ -893,20 +1125,30 @@ input::placeholder::first-letter {
   width: 100%;
 }
 
+.nft-card-list {
```
I see in the design we have 3 card on the row. Could you please recheck this moment?
<img width="962" alt="image" src="https://github.com/user-attachments/assets/39e73de6-09f1-4f0a-a5d4-145619e7b0a6">

annasvrv replied:
#fixed

alex-koshara reviewed (commented):
Amazing! Very good job, keep it up!

From what I saw, this is more of a recommendation on how to improve your code!
I really liked it:
1) you use grids and flex
2) wrap the elements correctly in the padding from all sides
3) you use different ways to indent

What would I recommend to improve in the future:
1) Use padding to the maximum where possible, and only then margin. Since we can't inspect margin through dev tools, but padding can
2) Try to avoid using fixed height for elements as much as possible, only if necessary. This will allow us to avoid errors when the content is overflowing.
3) Pay attention to margins that are not multiples of 2 or 5. Designers are also people and can make mistakes on 1px very easily. For example, I saw a gap on 79px, it also looks like a design error
4) try to make sure that the content does not fall out of the parent. This is precisely due to the height restriction
5) as an additional task, you can fix the code so that nothing breaks when the screen is reduced

Amazing, amazing job! Keep up the good work, it was very nice to check out such diligent work.

annasvrv committed:
Fixed issues.

Changed margin to paddings where possible.
Checked fixed heights and widths. Fixed where possible.
Added grids.
Created some adaptive layouts.

annasvrv committed:
Removed container class.
Moved link tag outside nft-card to make whole card a link.
Renamed some classes for grid styles.

annasvrv committed:
PR fixes.

Changed and added grid layouts.
Adapted some styles to small screens.
Changed margins to padding where possible.
Removed container class.
Moved link tag outside nft-card to make whole card a link.
Cleaned up.

annasvrv committed:
Removed unecessary style

annasvrv committed:
Road map progress bar - removed height restriction

annasvrv committed:
Indented hero-title header

annasvrv committed:
Roadmap progress bar fixed - created checkbox form

annasvrv committed:
Roadmap progress bar fixed - added progress classes and styles

annasvrv committed:
fixed some css according to css validator

annasvrv committed:
Added favicon 192

alex-koshara commented on css/style.css:
```diff
@@ -114,61 +116,63 @@ button {
+  position: sticky;
+  padding-top: 30px;
+  padding-bottom: 30px;
   top: 0px;
```
Just remember, 0 doesn't have measures

alex-koshara commented on css/style.css:
```diff
@@ -255,42 +262,109 @@ button {
+  display: grid;
+  grid-template-columns: repeat(2, 1fr);
+  grid-template-rows: 1fr 0.4fr;
+  grid-template-areas:
```
Nice work with grids! 
If you would like do it more adaptively, then try to do it so that when you add more properties and values, nothing breaks and they appear

alex-koshara commented:
A very wonderful job! Please pay attention in the future to the fact that zero has no units of measurement, I saw 9 such records. And about the overflow of content, it's like an addition, try to check for it, too, if possible.  If you have any questions, ask them freely
